framework/src/materials/DerivativeFunctionMaterialBase.C

```diff
@@ -56,13 +56,13 @@ DerivativeFunctionMaterialBaseTempl<is_ad>::DerivativeFunctionMaterialBaseTempl(
   for (unsigned int i = 0; i < _nargs; ++i)
   {
     // first derivatives (TODO: declareGenericPropertyDerivative)
-    _prop_dF[i] = &this->template declarePropertyDerivative<Real>(_F_name, _arg_names[i]);
+    _prop_dF[i] = &this->template declarePropertyDerivative<Real, is_ad>(_F_name, _arg_names[i]);
```

Comment:
Oh I see, the comment right above calls this out. There is no declarePropertyDerivative<Real, true> yet...UG so close in getting centipede fully AD...

Reply:
Ah. I don't know what's going on, seems like `&getMaterialPropertyDerivative<Real, true>` should work, but it doesn't pull the derivatives.

Reply:
Wait, what?

Reply:
What you can't understand my late night ramblings? :)

I have a trying-to-be converted to AD kernel that takes derivative from the material created by `DerivativeSumMaterial`. I tried taking the derivatives via `getMaterialPropertyDerivative<Real, true>(_derivative_sum_material_name, _var.name())`, but that always returns zero. When I use `getADMaterialProperty<Real>(derivativePropertyNameFirst(_linked_mat_names[i], _var.name()))` (based off of `ADACInterface`), I get `Material property 'df_name/dvar', requested by 'somekernel' is not defined on block 0, so it seems like `DerivativeSumMaterial` is not creating all the derivatives it needs. If this is too confusing, I can send you a source file...

Reply:
Uuuuaahhhh, how does any of this work?! How did the unit test I wrote even work?! I'll get on this.

Reply:
Actually, my problem isn't with `ADDerivativeSumMaterial`, it's with trying to pull derivatives as I explained above. I was able to double check this by just making a ADDerivativeParsedMaterial` that did the same thing as `ADDerivativeSumMaterial`, it seems like it is deeper with how derivatives are being pulled?

Reply:
OH I got it!! When doing `derivativePropertyNameFirst`, It doesn't do a name look up like getMaterialPropertyDerivative does!

Reply:
You may be well on your way here anyways, but just a note that `getADMaterialProperty<Real>(derivativePropertyNameFirst("some_derivative_parsed_material", "some_var"))` will return an error instead of a zero if "some_derivative_parsed_material" doesn't depend on "some_var".

Reply:
Ok, after taking a closer look, the `TODO: declareGenericPropertyDerivative` should have been removed, because `declarePropertyDerivative<Real, is_ad>` does just that. I'll switch to looking at the defaultMaterialProperty stuff

Reply:
Yeah exactly. It mostly works, just the small helper details

Reply:
Oooof...that startup is suppppper slow now with derivatives of AD...

Reply:
every time or just the first compilation?

Reply:
Well first time, and then JIT kicks in...but remember I'm running thousands of these things...stochastic tools would help

Reply:
I am running with sparse matrix...would it be better to be non-sparse?

Reply:
AD with the derivatives is unusable for the very large problems